app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/integration/IntegrationOverviewHelper.java

```diff
@@ -84,48 +86,50 @@ static IntegrationOverview toCurrentIntegrationOverview(final Integration integr
                 .collect(Collectors.toList()));
         }
 
-        IntegrationDeployment deployed = null;
+        final ListResult<IntegrationDeployment> deployments = dataManager.fetchAll(IntegrationDeployment.class,
+                new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance());
+
         final List<IntegrationDeployment> activeDeployments = new ArrayList<>();
-        for (final IntegrationDeployment deployment : dataManager.fetchAll(IntegrationDeployment.class,
-            new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance())) {
+        for (final IntegrationDeployment deployment : deployments) {
             builder.addDeployment(IntegrationDeploymentOverview.of(deployment));
 
-            final IntegrationDeploymentState currentState = deployment.getCurrentState();
-            if (currentState == IntegrationDeploymentState.Published) {
-                deployed = deployment;
-                builder.deploymentVersion(deployment.getVersion());
-            }
-
-            if (currentState != IntegrationDeploymentState.Unpublished) {
-                // the bet is that any integration that the user wanted to
-                // publish
-                // will have it's status != Unpublished, the reason why we can't
-                // look at the last deployment is because users can choose to
-                // deploy
-                // previous deployments, so we bet that all the Unpublished
-                // integrations are not the ones that the user don't hold the
-                // current state
-                builder.targetState(deployment.getTargetState());
-                builder.currentState(currentState);
+            if (deployment.getCurrentState().equals(IntegrationDeploymentState.Pending) ||
+                deployment.getCurrentState().equals(IntegrationDeploymentState.Published)) {
                 activeDeployments.add(deployment);
             }
         }
 
-        if (deployed != null) {
-            builder.isDraft(computeDraft(integration, deployed.getSpec()));
+        Optional<IntegrationDeployment> maybePublished = activeDeployments.stream()
+                  .filter(deployment -> deployment.getCurrentState().equals(IntegrationDeploymentState.Published))
+                  .findAny();
```

Comment:
```suggestion
                  .findFirst();
```

`findAny()` is [non-deterministic with regards to ordering](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#findAny--), I think we wan't the first published deployment (and I think that should be the last deployment for that integration)

Reply:
The user can also redeploy a previous deployment so it is not always the last deployment in the list which is in published state.

Though there should be only one published deployment in that list at the same time so findFirst is fine